Kernel/Heap/kmalloc.cpp

```diff
@@ -159,6 +159,9 @@ struct KmallocGlobalData {
 
     void* allocate(size_t size)
     {
+        // During early boot, we don't have a Thread yet; in fact, we might be allocating its memory right now!
+        if (Processor::current_thread() != nullptr)
```

Comment:
We already look up the current thread in `kmalloc(..)` so it seems like we should just move these ch checks there to avoid the double TLS fetch.